src/FirebirdSql.EntityFrameworkCore.Firebird.Tests/Scaffolding/ScaffoldingTests.cs

```diff
@@ -15,21 +15,117 @@
 
 //$Authors = Jiri Cincura (jiri@cincura.net)
 
+using System.Linq;
+using System.Threading.Tasks;
 using FirebirdSql.EntityFrameworkCore.Firebird.Scaffolding.Internal;
+using Microsoft.EntityFrameworkCore;
 using Microsoft.EntityFrameworkCore.Scaffolding;
 using NUnit.Framework;
 
 namespace FirebirdSql.EntityFrameworkCore.Firebird.Tests.Scaffolding;
 #pragma warning disable EF1001
 public class ScaffoldingTests : EntityFrameworkCoreTestsBase
 {
+	class ScaffoldingContext : FbTestDbContext
+	{
+		public ScaffoldingContext(string connectionString)
+			: base(connectionString)
+		{ }
+
+		protected override void OnTestModelCreating(ModelBuilder modelBuilder)
+		{
+			base.OnTestModelCreating(modelBuilder);
+		}
+	}
+
 	[Test]
 	public void JustCanRun()
 	{
 		var modelFactory = GetModelFactory();
 		Assert.DoesNotThrow(() => modelFactory.Create(Connection, new DatabaseModelFactoryOptions()));
 	}
 
+	[Test]
+	public async Task ReadsNullableCorrect()
+	{
+		// Arrange
+		var tableName = "TEST_READS_IS_NULL_FROM_DOMAIN";
+		var columnNameNoDomainNull = "NO_DOMAIN_NULL";
+		var columnNameNoDomainNotNull = "NO_DOMAIN_NOT_NULL";
+		var columnNameDomainNull = "DOMAIN_NULL";
+		var columnNameDomainNotNull = "DOMAIN_NOT_NUL";
+
+		using var commandDomainNull = Connection.CreateCommand();
+		commandDomainNull.CommandText = "create domain DOMAIN_NULL as INTEGER";
+		await commandDomainNull.ExecuteNonQueryAsync();
+
+		using var commandDomainNotNull = Connection.CreateCommand();
+		commandDomainNotNull.CommandText = "create domain DOMAIN_NOT_NULL as INTEGER NOT NULL";
+		await commandDomainNotNull.ExecuteNonQueryAsync();
+
+		using var commandTable = Connection.CreateCommand();
+		commandTable.CommandText = $"create table {tableName} ({columnNameNoDomainNull} INTEGER, {columnNameNoDomainNotNull} INTEGER NOT NULL, {columnNameDomainNull} DOMAIN_NULL, {columnNameDomainNotNull} DOMAIN_NOT_NULL)";
+		await commandTable.ExecuteNonQueryAsync();
```

Comment:
All database objects created should be removed in `TearDown`.

Reply:
Going thru other tests:
- [QueryTests](https://github.com/FirebirdSQL/NETProvider/blob/f169563ec05067c55974d2dfe09ff4f64cfd05d9/src/EntityFramework.Firebird.Tests/QueryTests.cs#L50) / [InsertTests](https://github.com/FirebirdSQL/NETProvider/blob/f169563ec05067c55974d2dfe09ff4f64cfd05d9/src/FirebirdSql.EntityFrameworkCore.Firebird.Tests/EndToEnd/InsertTests.cs#L53) and [UpdateTests](https://github.com/FirebirdSQL/NETProvider/blob/f169563ec05067c55974d2dfe09ff4f64cfd05d9/src/FirebirdSql.EntityFrameworkCore.Firebird.Tests/EndToEnd/UpdateTests.cs#L56) do not remove its tables (or I missed out the important point)
- [FbBatchCommandTests](https://github.com/FirebirdSQL/NETProvider/blob/f169563ec05067c55974d2dfe09ff4f64cfd05d9/src/FirebirdSql.Data.FirebirdClient.Tests/FbBatchCommandTests.cs#L50) creates tables in SetUp and removes them in TearDown
- [FbBatchCommandTests](https://github.com/FirebirdSQL/NETProvider/blob/f169563ec05067c55974d2dfe09ff4f64cfd05d9/src/FirebirdSql.Data.FirebirdClient.Tests/FbCommandTests.cs#L260) uses in some cases try / finally 

Initially I had the idea to introduce new methods for that purpose:

``` csharp
public abstract class ScaffoldingTestsBase : FbTestsBase
{
	private readonly List<string> _tables = new();
	private readonly List<string> _domains = new();

	public ScaffoldingTestsBase()
		: base(FbServerType.Default, false, FbWireCrypt.Enabled, false)
	{ }


	protected async Task CreateTable(string name, params string[] fields)
	{
		using var command = Connection.CreateCommand();
		command.CommandText = $"create table {name} ({string.Join(", ", fields)})";
		await command.ExecuteNonQueryAsync();
		_tables.Add(name);
	}

	protected async Task CreateDomain(string name, string type)
	{
		using var command = Connection.CreateCommand();
		command.CommandText = $"create domain {name} as {type}";
		await command.ExecuteNonQueryAsync();
		_domains.Add(name);
	}

	[TearDown]
	public override async Task TearDown()
	{
		foreach (var table in _tables)
		{
			var command = Connection.CreateCommand();
			command.CommandText = $"drop table {table}";
			await command.ExecuteNonQueryAsync();
		}
		_tables.Clear();
		foreach (var domain in _domains)
		{
			var command = Connection.CreateCommand();
			command.CommandText = $"drop domain {domain}";
			await command.ExecuteNonQueryAsync();
		}
		_domains.Clear();
		await base.TearDown();
	}
}
```

But i am unsure if this is the right aproach here:
- These method would theoretically belong to `FbTestsBase` since creating tables or domains is not only required in scaffolding.
- The methods are not thread save which means when someone in furture starts to run tests with [Parallelizable] this will probably result in ungly errors.

Long story short: What is the best practice to remove database objects?

Reply:
> The methods are not thread save which means when someone in furture starts to run tests with [Parallelizable] this will probably result in ungly errors.

Parallel (provider) testing with databases is can of worms. I have no plans to do parallel testing in foreseeable future, mostly because that would mean reworking lot of tests.

> Long story short: What is the best practice to remove database objects?

As you can see there's no consistent story (tests were created in different times with slightly varying setups). I think we can leave it as is, and when the time comes, the creation and cleanup will be unified.